api/nakadi-event-bus-api.yaml

```diff
@@ -1083,6 +1139,99 @@ definitions:
         items:
           $ref: '#/definitions/Event'
 
+  Subscription:
+    description: |
+      Subscription is a high level consumption unit. Subscriptions allow applications to easily scale the number of
+      clients by managing consumed event offsets and distributing load between instances.
+      The key properties that identify subscription are 'owning_application', 'event_types' and 'use_case'.
+      It's not possible to have two different subscriptions with these properties being the same.
+    properties:
+      id:
+        type: string
+        description: |
+          Id of subscription that was created. Is generated by Nakadi, should not be specified when creating
+          subscription.
+        format: uuid
+        example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+      created_at:
+        type: string
+        description: |
+          Timestamp of creation of the subscription. Is generated by Nakadi, should not be specified when creating
+          subscription.
+        format: date-time
+        example: '1996-12-19T16:39:57-08:00'
+      owning_application:
+        type: string
+        example: 'price-service'
+        description: The id of application owning the subscription.
+      event_types:
+        type: array
+        items:
+          type: string
+        description: |
+          EventTypes to subscribe to.
+          The order is not important. Subscriptions that differ only be the order of EventTypes will be considered
+          the same and will have the same id.
+          * Currently only subscription to a single EventType is supported. Subscriptions with more than one EventType
+          in event_types property will be rejected.
+      use_case:
+        type: string
+        example: 'read-product-updates'
+        description: |
+          The value describing the use case of this subscription.
+          In general that is an additional identifier used to differ subscriptions having the same owning_application
+          and event_types.
+        default: 'none'
+      read_from:
+        type: string
+        enum:
+          - BEGIN
+          - END
+        description: |
+          Position to start reading events from. BEGIN - read from the oldest available event. END - read from the most
+          recent offset. Will be applied when the first client for this subscription starts consuming.
```

Comment:
I see your point but there are trade offs for each solution.
If we fetch offsets when subscription is created then it also has some issues:

Example 1, user creates subscription and defines that he wants to read from BEGIN. Nakadi fetches the earliest available offset (let's say 43809) and defines it as the one to start streaming from. 10 hours after creation of subscription user creates first stream and starts reading events. Offset 43809 is not available any more so the stream is started from offset 44102. So the BEGIN will anyway be the earliest available offsets when first client started.

Example 2, user creates subscription and defines that he wants to read from END. Nakadi fetches the earliest available offset (let's say 57098) and defines it as the one to start streaming from. When first client connects and starts reading that is not a real end any more, there were few hundreds of events pushed between subscription creation and start of stream.
Another possibility here is that first stream was created one week after subscription creation and at that time event END offset is expired. So the stream will be started from the earliest offset that is currently available

That's why I think that defining BEGIN and END relatively to subscription creation will be quite confusing for the user.